benchmarks/helics/ActionMessageBenchmarks.cpp

```diff
@@ -10,138 +10,128 @@ SPDX-License-Identifier: BSD-3-Clause
 
 using namespace helics;
 
-static void BM_AM_toString (benchmark::State &state)
+static void BM_AM_toString(benchmark::State& state)
 {
-    ActionMessage obj (CMD_REG_FED);
+    ActionMessage obj(CMD_REG_FED);
     obj.name = "the name of the federate is really long";
-    obj.setStringData ("this is a new string to add to the string data");
+    obj.setStringData("this is a new string to add to the string data");
     std::string load;
-    load.reserve (500);
-    for (auto _ : state)
-    {
-        obj.to_string (load);
+    load.reserve(500);
+    for (auto _ : state) {
```

Comment:
@kdheepak @nightlark  what do you think of the Linux mode for braces.    I don't like attach mode for functions and classes but mostly it is Ok for if/for/while loops

Reply:
I prefer `Attach` for everything. It's easier to differentiate between declarations and definitions as well this well, since by the time you read to the end of the line you know if this is a function declaration or definition. It's also consistent with formatting recommendations from other curly braces based language. 

Is there a reason you don't like `Attach` for functions and classes?

Reply:
The main reason is whitespace.   When looking through a cpp file that has a 1000 or more lines in it The lack of whitespace from attach mode makes it harder to scan through when you need to edit.  

Having the brackets on separate lines isolates function definitions making them distinct and easy to scan through when working with large cpp files.  codes using attach just look very cluttered to me and harder to visually parse when working with them.  This is mainly true of functions and methods and to a lesser extent with classes.  larger functions also need additional whitespace lines if attach is used, probably will also with Linux mode but there are some other advantages to attaching the bracket for small loops and conditions.

Reply:
If there isn't an existing style for the code, attach (probably was from Java or whatever some C++ book used, and it's similar to go/rust). No preference for one or the other though.

Reply:
What do you think about `Allman` and `GNU` styles?

Reply:
Allman was what we had been using.   and I can't say I really care for the indented brackets of GNU style.  

I think when it comes down to it functions are the main point at which Attach creates issues for me on larger files.  

Stroustrup is a possibility as well.   if you wanted classes to have the attached brackets.

Reply:
Same here with the indented brackets of GNU style. Linux mode looks pretty good.